com.fyralabs.Exponential.yaml

```diff
@@ -0,0 +1,56 @@
+app-id: com.fyralabs.Exponential
+
+runtime: org.freedesktop.Platform
+runtime-version: "23.08"
+sdk: org.freedesktop.Sdk
+
+base: org.electronjs.Electron2.BaseApp
+base-version: "23.08"
+
+command: com.fyralabs.Exponential
+separate-locales: false
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  # - --socket=wayland
+  # - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=network
+  - --env=ELECTRON_TRASH=gio
+  - --device=dri
+  - --talk-name=com.canonical.Unity
+
+modules:
+  - name: Exponential
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 ${FLATPAK_ID}.sh /app/bin/${FLATPAK_ID}
+      - install -Dm644 ${FLATPAK_ID}.png /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm644 ${FLATPAK_ID}.metainfo.xml /app/share/appdata/${FLATPAK_ID}.metainfo.xml
+      - install -Dm644 ${FLATPAK_ID}.desktop /app/share/applications/${FLATPAK_ID}.desktop
+      - mkdir /app/exponential
+      - unzip electron.zip -d /app/exponential
+      - rm -rf /app/exponential/resources/*
+      - unzip -j -d "/app/exponential/resources" linear.zip 'Linear.app/Contents/Resources/*'
+      - patch-desktop-filename ${FLATPAK_DEST}/exponential/resources/app.asar
+    sources:
+      - type: file
+        url: https://download.todesktop.com/200315glz2793v6/Linear%201.27.0%20-%20Build%202408152uraifa26-x64-mac.zip
```

Comment:
why does the build say "mac"? This is not macOS.

Reply:
We use the .asar file (which is universal) from the macOS build.

Reply:
what is the relation of todesktop.com with fyralabs.com ?

Reply:
the situation is a bit confusing, but I can try to explain:
fyralabs.com is a domain I control, which I'm using for the repack, upstream wanted us to use distinct branding from "Linear", which is why the app is called "Exponential", rather than "Linear"
linear.app is the product that is being repacked
todesktop.com is a service that Linear uses to actually build the desktop application (and bundle), and acts as a CDN for the build artifacts

Reply:
So linear.app is the upstream of the main app here ? So you are submitting it under com.fryalabs because they asked you to ?

> upstream wanted us to use distinct branding from "Linear", which is why the app is called "Exponential", rather than "Linear"

Is this simply a rebrand or a fork ?

Can we get some kind of confirmation on that ? If not this should be called `app.linear.foo.bar`

Reply:
Apps must use the original branding, upstream's app ID. If they want it "unofficial" there are other ways to do that like having a note on the description and the unverified badge.

If you submit it under your domain, it allows someone else to verify someone else's apps

Reply:
> you are submitting it under com.fryalabs because they asked you to

Correct, they seemed to not want any of the app metadata (besides a mention in the description) tied to them.

> Is this simply a rebrand or a fork ?

rebrand

> Apps must use the original branding, upstream's app ID.

I'll ask again, I originally suggested the idea of just having a note in the description, but they didn't think that was sufficient.

Reply:
This make the license dubious at best. This is not open source and you want us to believe it's okay based on a screenshot of a random discord conversation.

Reply:
Marking it as blocked because:

a) we don't know if branding modifications are allowed as it's under a proprietary license

b) the ID does not follow the requirements: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain-or-repository

> It must be a domain directly related to the original project and the author/developer/project of the application must have control over the domain portion of the ID

Even if the modification was allowed, the ID must be `app.linear`, you can make the user visible parts like metainfo, desktop file, icon use the new branding.